open-api/rest-catalog-open-api.yaml

```diff
@@ -134,9 +134,20 @@ paths:
     post:
       tags:
         - OAuth2 API
-      summary: Get a token using an OAuth2 flow
+      summary: Get a token using an OAuth2 flow (DEPRECATED for REMOVAL)
+      deprecated: true
       operationId: getToken
       description:
+        The `oauth/tokens` endpoint is **DEPRECATED for REMOVAL**. It is not recommended to _not_
+        implement this endpoint, unless you are fully aware of the potential security implications.
+        All clients are encouraged to explicitly set the configuration property `oauth2-server-uri`
+        to the correct OAuth endpoint.
+        See [Security improvements in the Iceberg REST specification](https://github.com/apache/iceberg/issues/10537)
```

Comment:
I think instead of linking to the GitHub issue, we probably want to add a dedicated section explaining OAuth2 usage, maybe in `open-api/security/oauth2.md`. Should we do it as a part of this PR? I am okay either way.

Reply:
Not sure TBH. Since the whole oauth stuff is eventually removed from the spec, I think it's okay to leave the link to the issue.